src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoHealthDataDocumentationDao.java

```diff
@@ -0,0 +1,132 @@
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import org.joda.time.DateTime;
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.HealthDataDocumentationDao;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.HealthDataDocumentation;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Resource;
+import java.util.List;
+
+@Component
+public class DynamoHealthDataDocumentationDao implements HealthDataDocumentationDao {
+    private DynamoDBMapper mapper;
+
+    /** DynamoDB mapper for the HealthDataDocumentation table, configured by Spring. */
+    @Resource(name = "healthDataDocumentationDbMapper")
+    public void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public HealthDataDocumentation createOrUpdateDocumentation(@Nonnull HealthDataDocumentation documentation) {
+        HealthDataDocumentation dynamoDocumentation = (DynamoHealthDataDocumentation) documentation;
+
+        if (dynamoDocumentation.getIdentifier() == null) {
+            // Documentation doesn't have ID assigned yet (new documentation). Create ID and assign it.
+            dynamoDocumentation.setIdentifier(BridgeUtils.generateGuid());
+
+            // Update created on attributes.
+            dynamoDocumentation.setCreatedOn(DateTime.now());
+        } else {
+            // Update modified on attributes.
+            dynamoDocumentation.setModifiedOn(DateTime.now());
+        }
+
+        // Save to DynamoDB.
+        mapper.save(dynamoDocumentation);
+        return dynamoDocumentation;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void deleteDocumentationForParentId(@Nonnull String parentId) {
+        // Query the records that we need to delete.
+        DynamoHealthDataDocumentation key = new DynamoHealthDataDocumentation();
+        key.setParentId(parentId);
+
+        DynamoDBQueryExpression<DynamoHealthDataDocumentation> query =
+                new DynamoDBQueryExpression<DynamoHealthDataDocumentation>().withHashKeyValues(key);
+        List<DynamoHealthDataDocumentation> docsToDelete = queryHelper(query);
+
+        // Batch delete.
+        if (!docsToDelete.isEmpty()) {
+            List<DynamoDBMapper.FailedBatch> failures = mapper.batchDelete(docsToDelete);
+            BridgeUtils.ifFailuresThrowException(failures);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void deleteDocumentationForIdentifier(@Nonnull String identifier, @Nonnull String parentId) {
```

Comment:
Minor: The order of arguments should be parentId, identifier. The general convention is we use the more general category before the more specific category. (At least when there is a clear hierarchy.)